src/libtiled/mapobject.cpp

```diff
@@ -209,16 +209,40 @@ void MapObject::flip(FlipDirection direction)
             mCell.setFlippedVertically(!mCell.flippedVertically());
     }
 
-    if (!mPolygon.isEmpty()) {
-        const QPointF center2 = mPolygon.boundingRect().center() * 2;
+    //creating flipmatrix
+    QMatrix flipMatrix;
```

Comment:
The documentation of QMatrix says "This class is obsolete. It is provided to keep old source code working. We strongly advise against using it in new code."

Please use QTransform instead.